libsolidity/codegen/ir/IRGenerationContext.cpp

```diff
@@ -104,6 +148,14 @@ string IRGenerationContext::variable(Expression const& _expression)
 	string var = "expr_" + to_string(_expression.id());
 	if (size == 1)
 		return var;
+	else if (
+		auto const* arrayType = dynamic_cast<ReferenceType const*>(_expression.annotation().type);
```

Comment:
We should find a generic way to solve this. I hope you agree that names are much better than numbers.

Reply:
The most generic solution I guess would be to have the types themselves return not only their ``sizeOnStack()`` but a list of preferred names for their stack slots...

Reply:
Should I go for that or do you have something better in mind?

Reply:
If it is a small change, we can do it. Then please use a default list of `_1, _2` for all types except for those we already defined names for.

Reply:
Ok, I'll write up something like that in a separate commit.